examples/save_course_slides.py

```diff
@@ -0,0 +1,142 @@
+# Run with Python3
+# Saves all presentations from course in pdfs and arranges them into folders
+# BeautifulSoup is required, just do:  pip install beautifulsoup4
+
+import requests
+import sys
+import os
+from bs4 import BeautifulSoup
+import urllib.request
+import urllib.response
+
+client_id = "..."
+client_secret = "..."
+api_host = "https://stepik.org/api/"
+
+
+# get a token
+def auth():
+    auth = requests.auth.HTTPBasicAuth(client_id, client_secret)
+    response = requests.post('https://stepik.org/oauth2/token/',
+                             data={'grant_type': 'client_credentials'},
+                             auth=auth)
+    global token
+    token = response.json().get('access_token', None)
+    if not token:
+        print('Unable to authorize with provided credentials')
+        exit(1)
+
+
+# get 1 json object
+def fetch_object(obj_class, obj_id):
+    api_url = "{}/{}s/{}".format(api_host, obj_class, obj_id)
+    response = requests.get(api_url, headers={'Authorization': 'Bearer ' + token}).json()
+    return response["{}s".format(obj_class)][0]
+
+
+# get json-objects with ids
+def fetch_objects(obj_class, ids):
+    objs = []
+    step_size = 30
+    for i in range(0, len(ids), step_size):
+        cur_ids = ids[i:i + step_size]
+        api_url = "{}/{}s/?".format(api_host, obj_class)
+        for id in cur_ids:
+            api_url += "ids[]={}&".format(id)
+        response = requests.get(api_url, headers={'Authorization': 'Bearer ' + token}).json()
+        objs += response["{}s".format(obj_class)]
+    return objs
+
+
+# convert name of section into proper name folder
+def replace_characters(text):
+    text = text.replace(":", " -")
+    text = text.replace("?", " ")
+    return text
+
+
+# create folder of course with readme.html about course
+def create_info_dir(title, workload, summary):
+    current_path = os.path.dirname(os.path.abspath(__file__))
+    global CURRENT_PATH
+
+    current_path = os.path.join(current_path, title)
+    CURRENT_PATH = current_path
+    if not os.path.exists(current_path):
+        os.makedirs(current_path)
+        with open(os.path.join(current_path, "readme.html"), "w") as info_file:
+            print(
+                "<h1>" + title + "</h1>" + "<p><b>Нагрузка</b>: " + workload + "</p>" + "<b>Коротко о курсе</b>: " + summary,
+                file=info_file)
+    else:
+        print("folder already exists")
+
+# create folder for section
+def create_folder(title):
+    if not os.path.exists(os.path.join(CURRENT_PATH, title)):
+        os.makedirs(os.path.join(CURRENT_PATH, title))
+
+
+# download pdf
+def download_file(link, path):
+    slides = urllib.request.urlopen(link)
```

Comment:
we already using `requests` , so there is no need to use urllib. `requests` library is preferable way to manage all http requests